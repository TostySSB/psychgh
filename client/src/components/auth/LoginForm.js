import React from 'react';
import OktaAuth from '@okta/okta-auth-js';
import { withAuth } from '@okta/okta-react';
import { Button, FormGroup, FormControl, ControlLabel } from "react-bootstrap";
import "../../Views/Login.css";
export default withAuth(
  class LoginForm extends React.Component {
    constructor(props) {
      super(props);
      this.state = {
        sessionToken: null,
        error: null,
        username: '',
        password: ''
      };

      this.oktaAuth = new OktaAuth({ url: props.baseUrl });

      this.handleSubmit = this.handleSubmit.bind(this);
      this.handleUsernameChange = this.handleUsernameChange.bind(this);
      this.handlePasswordChange = this.handlePasswordChange.bind(this);
    }

    handleSubmit(e) {
      e.preventDefault();
      this.oktaAuth
        .signIn({
          username: this.state.username,
          password: this.state.password
        })
        .then(res =>
          this.setState({
            sessionToken: res.sessionToken
          })
        )
        .catch(err => {
          this.setState({ error: err.message });
          console.log(err.statusCode + ' error', err);
        });
    }

    handleUsernameChange(e) {
      this.setState({ username: e.target.value });
    }

    handlePasswordChange(e) {
      this.setState({ password: e.target.value });
    }

    render() {
      if (this.state.sessionToken) {
        this.props.auth.redirect({ sessionToken: this.state.sessionToken });
        return null;
      }

      const errorMessage = this.state.error ? (
        <span className="error-message">{this.state.error}</span>
      ) : null;

      return (
        <div className="Login">
          <form onSubmit={this.handleSubmit}>
            {errorMessage}
            <FormGroup controlId="username" bsSize="large">
              <ControlLabel>Email:</ControlLabel>
              <FormControl
                id="username"
                type="text"
                value={this.state.username}
                onChange={this.handleUsernameChange}
              />
            </FormGroup>

            <FormGroup controlId="password" bsSize="large">
              <ControlLabel>Password:</ControlLabel>
              <FormControl
                id="password"
                type="password"
                value={this.state.password}
                onChange={this.handlePasswordChange}
              />
            </FormGroup>
            <Button
            block
            bsSize="large"
            type="submit"
            >
            Login
            </Button>
          </form>
        </div>
      );
    }
  }
);