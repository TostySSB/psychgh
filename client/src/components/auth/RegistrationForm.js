import React from 'react';
import OktaAuth from '@okta/okta-auth-js';
import { withAuth } from '@okta/okta-react';

import config from '../../app.config';
import {
  HelpBlock,
  FormGroup,
  FormControl,
  ControlLabel
} from "react-bootstrap";
import LoaderButton from "../../components/LoaderButton";
import "../../Views/Signup.css";

export default withAuth(
  class RegistrationForm extends React.Component {
    constructor(props) {
      super(props);
      this.state = {
        firstName: '',
        lastName: '',
        email: '',
        password: '',
        sessionToken: null
      };
      this.oktaAuth = new OktaAuth({ url: config.url });
      this.checkAuthentication = this.checkAuthentication.bind(this);
      this.checkAuthentication();

      this.handleSubmit = this.handleSubmit.bind(this);
      this.handleFirstNameChange = this.handleFirstNameChange.bind(this);
      this.handleLastNameChange = this.handleLastNameChange.bind(this);
      this.handleEmailChange = this.handleEmailChange.bind(this);
      this.handlePasswordChange = this.handlePasswordChange.bind(this);
    }

    async checkAuthentication() {
      const sessionToken = await this.props.auth.getIdToken();
      if (sessionToken) {
        this.setState({ sessionToken });
      }
    }

    componentDidUpdate() {
      this.checkAuthentication();
    }

    handleFirstNameChange(e) {
      this.setState({ firstName: e.target.value });
    }
    handleLastNameChange(e) {
      this.setState({ lastName: e.target.value });
    }
    handleEmailChange(e) {
      this.setState({ email: e.target.value });
    }
    handlePasswordChange(e) {
      this.setState({ password: e.target.value });
    }

    handleSubmit(e) {
      e.preventDefault();
      fetch('/api/users', {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json'
        },
        body: JSON.stringify(this.state)
      })
        .then(user => {
          this.oktaAuth
            .signIn({
              username: this.state.email,
              password: this.state.password
            })
            .then(res =>
              this.setState({
                sessionToken: res.sessionToken
              })
            );
        })
        .catch(err => console.log);
    }

    render() {
      if (this.state.sessionToken) {
        this.props.auth.redirect({ sessionToken: this.state.sessionToken });
        return null;
      }

      return (
        <div className="Signup">
          <form onSubmit={this.handleSubmit}>
          <FormGroup controlId="email" bsSize="large">
            <ControlLabel>Email:</ControlLabel>
              <FormControl
                type="email"
                id="email"
                value={this.state.email}
                onChange={this.handleEmailChange}
              />
            </FormGroup>
            <FormGroup controlId="firstName" bsSize="large">
              <ControlLabel>First Name:</ControlLabel>
              <FormControl
                type="text"
                id="firstName"
                value={this.state.firstName}
                onChange={this.handleFirstNameChange}
              />
            </FormGroup>
            <FormGroup controlId="lastName" bsSize="large">
              <ControlLabel>Last Name:</ControlLabel>
              <FormControl
                type="text"
                id="lastName"
                value={this.state.lastName}
                onChange={this.handleLastNameChange}
              />
            </FormGroup>
            <FormGroup controlId="password" bsSize="large">
              <ControlLabel>Password:</ControlLabel>
              <FormControl
                type="password"
                id="password"
                value={this.state.password}
                onChange={this.handlePasswordChange}
              />
            </FormGroup>
            <input type="submit" id="submit" value="Register" />
          </form>
        </div>
      );
    }
  }
);