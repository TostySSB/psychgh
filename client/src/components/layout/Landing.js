import React, { Component } from "react";
import HomePageGrid from '../UI/grids/HomePageGrid';
class Landing extends Component {
  render() {
    return (
      <div>
        <div style={{ paddingTop: "10%", paddingBottom: "10%" }} className="container valign-wrapper" >
        <div className="row">
          <div className="col s12 center-align">
            <h4>
              <b>Welcome</b> to Psych432, a work in progress app designed to make the lives of patients and practitioners easier.
            </h4>
            <p className="flow-text grey-text text-darken-1">
              Proudly built using the MERN stack.
            </p>
          </div>
        </div>
      </div>
        <HomePageGrid></HomePageGrid>
        <div className="col s12 center-align">
            <p className="flow-text grey-text text-darken-1">
              For assistance, please contact us at: tempEmailName@gmail.com
            </p>
          </div>
      </div>
      
    );
  }
}

export default Landing;
