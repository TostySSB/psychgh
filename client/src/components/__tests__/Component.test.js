import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Login} from '../auth/Login';
import {Register} from '../auth/Register';
import {Dashboard} from '../dashboard/Dashboard';
import Landing from '../layout/Landing';
import Navbar from '../layout/Navbar';

// Test all components for rendering without crashing
configure({adapter: new Adapter()});

// function setup() {
// 	const props = {
// 		addTodo: jest.fn()
// 	};
// 	const enzymeWrapper = shallow(<Landing />)
// 	return {props, enzymeWrapper};
// }

// Test layout components
describe('Test layout components', () => {
	it('Landing Renders', () => {
		// shallow(<Landing />);
		const LandingComponent = renderer.create(<Landing />).toJSON();
		expect(LandingComponent).toMatchSnapshot();
	});

	it('Navbar Renders', () => {
		shallow(<Navbar />)
	});
});

// Test 
