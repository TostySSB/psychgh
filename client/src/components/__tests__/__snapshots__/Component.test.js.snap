// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test layout components Landing Renders 1`] = `
<div>
  <div
    className="container valign-wrapper"
    style={
      Object {
        "paddingBottom": "10%",
        "paddingTop": "10%",
      }
    }
  >
    <div
      className="row"
    >
      <div
        className="col s12 center-align"
      >
        <h4>
          <b>
            Welcome
          </b>
           to Psych432, a work in progress app designed to make the lives of patients and practitioners easier.
        </h4>
        <p
          className="flow-text grey-text text-darken-1"
        >
          Proudly built using the MERN stack.
        </p>
      </div>
    </div>
  </div>
  <div
    className="HomePageGrid-root-1"
  >
    <div
      className="MuiGrid-container-2 MuiGrid-justify-xs-space-evenly-23"
      style={
        Object {
          "minHeight": "100vh",
        }
      }
    >
      <div
        className="MuiGrid-item-3"
      >
        <div
          className="MuiPaper-root-102 MuiPaper-elevation1-105 MuiPaper-rounded-103 MuiCard-root-101 GitHubCard-card-99"
        >
          <div
            className="MuiCardMedia-root-129 GitHubCard-media-100"
            style={
              Object {
                "backgroundImage": "url(\\"GHLogo.png\\")",
              }
            }
            title="GitHub Logo"
          />
          <div
            className="MuiCardContent-root-131"
          >
            <h2
              className="MuiTypography-root-132 MuiTypography-h4-147 MuiTypography-gutterBottom-159 MuiTypography-alignCenter-155"
            >
              This is an open source project developed by Troy Oster, Henry Soule, and George Engel. Psychgh is hosted on GitHub. Click below to view the source code for this project.
            </h2>
          </div>
          <div
            className="MuiCardActions-root-168"
          >
            <a
              className="MuiButtonBase-root-197 MuiButton-root-171 MuiButton-text-173 MuiButton-textPrimary-174 MuiButton-flat-176 MuiButton-flatPrimary-177 MuiButton-sizeLarge-195 MuiButton-fullWidth-196 MuiCardActions-action-170"
              href="https://github.com/TostySSB/psychgh"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="MuiButton-label-172"
              >
                <h2
                  className="MuiTypography-root-132 MuiTypography-h5-148 MuiTypography-colorPrimary-162 MuiTypography-alignCenter-155"
                >
                  Learn More
                </h2>
              </span>
              <span
                className="MuiTouchRipple-root-202"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-3"
      >
        <div
          className="MuiPaper-root-102 MuiPaper-elevation1-105 MuiPaper-rounded-103 MuiCard-root-101 ReadDocsCard-card-200"
        >
          <div
            className="MuiCardMedia-root-129 ReadDocsCard-media-201"
            style={
              Object {
                "backgroundImage": "url(\\"readDocs.png\\")",
              }
            }
            title="ReadTheDocs Logo"
          />
          <div
            className="MuiCardContent-root-131"
          >
            <h2
              className="MuiTypography-root-132 MuiTypography-h4-147 MuiTypography-gutterBottom-159 MuiTypography-alignCenter-155"
            >
              Want to learn how to use, modify, and edit this project? Whether you are a developer, or simply a user, our documentation can help you. Click the link below to view the documentation for this project.
            </h2>
          </div>
          <div
            className="MuiCardActions-root-168"
          >
            <a
              className="MuiButtonBase-root-197 MuiButton-root-171 MuiButton-text-173 MuiButton-textPrimary-174 MuiButton-flat-176 MuiButton-flatPrimary-177 MuiButton-sizeLarge-195 MuiButton-fullWidth-196 MuiCardActions-action-170"
              href="https://psychgh.readthedocs.io/en/latest/"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="MuiButton-label-172"
              >
                <h2
                  className="MuiTypography-root-132 MuiTypography-h5-148 MuiTypography-colorPrimary-162 MuiTypography-alignCenter-155"
                >
                  Learn More
                </h2>
              </span>
              <span
                className="MuiTouchRipple-root-202"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
